www/src/components/IconsTable.jsx

```diff
@@ -1,27 +1,177 @@
+  style,
+}) => {
+  const {
+    iconsList, previewRef, setCurrentIcon, columnCount,
+  } = data;
+  const index = rowIndex * columnCount + columnIndex;
+  const iconName = iconsList[index];
+
+  const handleClick = () => {
+    setCurrentIcon(iconName);
+    if (previewRef.current) {
+      previewRef.current.scrollIntoView({ behavior: 'smooth', block: 'center' });
+    }
+  };
+
+  if (!iconName) {
+    return null;
+  }
+
+  return (
+    <>
+      <div
+        role="button"
+        key={iconName}
+        className="pgn-doc__icons-table__cell"
+        style={style}
+        onKeyPress={handleClick}
+        onClick={handleClick}
+        tabIndex={0}
+      >
+        <Icon
+          key={iconName}
+          src={IconComponents[iconName]}
+        />
+        <span className="pgn-doc__icons-table__cell-text">{iconName}</span>
+      </div>
+    </>
+  );
+};
+
+const IconsTable = () => {
+  const previewRef = React.useRef(null);
+  const [searchValue, setSearchValue] = useState('');
+  const [iconsList, setIconsList] = useState(ICON_NAMES);
+  const [currentIcon, setCurrentIcon] = useState(ICON_NAMES[0]);
+  const [showToast, setShowToast] = useState(false);
+  const currentIconImport = `import { ${currentIcon} } from '@edx/paragon/icons';`;
+
+  const copyToClipboard = (content) => {
+    navigator.clipboard.writeText(content);
+    setShowToast(true);
+  };
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedSetIconsList = useCallback(
+    debounce(setIconsList, 250, { leading: true }),
+    [],
+  );
+
+  useEffect(() => {
+    debouncedSetIconsList(
+      ICON_NAMES.filter(name => name.toLowerCase().includes(searchValue.toLowerCase())),
+    );
+  }, [searchValue, iconsList, debouncedSetIconsList]);
+
   return (
-    <div>
-      {Object.keys(IconComponents).map(iconName => (
-        <div key={iconName} className="d-flex mb-3 align-items-center">
+    <>
+      <SearchField
+        onSubmit={() => {}}
+        onChange={value => setSearchValue(value)}
+        placeholder="Search icons"
+      />
+      {currentIcon && (
+        <div className="pgn-doc__icons-table__preview" ref={previewRef}>
+          <h3
+            role="presentation"
+            className="rounded"
+            onClick={() => copyToClipboard(currentIcon)}
+          >
+            {currentIcon}
+          </h3>
           <Icon
-            className="mr-3"
-            key={iconName}
-            src={IconComponents[iconName]}
+            key={currentIcon}
+            src={IconComponents[currentIcon]}
           />
-          <h3 className="mb-0 mr-3 flex-grow-1 h5">{iconName}</h3>
-          <code className="bg-light px-2 py-1 rounded">
+          <code
+            role="presentation"
+            className="rounded"
+            onClick={() => copyToClipboard(currentIconImport)}
+          >
             <small>
-              {`import { ${iconName} } from '@edx/paragon/icons';`}
+              {currentIconImport}
```

Comment:
I'm thinking it might be a good idea to disclose that it is possible to copy this import string, without having to stumble on it. Maybe we show an inline/small `ContentCopy` icon to the right of the import statement that is the trigger to copy?

Reply:
Yes, I added `ContentCopy ` to the right and hover styles